### PR TITLE
Added hotkeys for toggling different stereoscopy modes

### PR DESCRIPTION
**Pull request by TurboK234**
Stereoscopy rocks, but I save it for 1) special occasions and 2) friend demonstrations. Both would benefit from a more streamlined ability to switch on (and off) stereoscopy, preferably with a hotkey.
This PR adds hotkeys for each stereoscopy mode. I didn't want to use cycling, as it's even uglier than using the GUI.
The downside of these hotkeys is that they mimic the GUI behaviour with anaglyph mode. Anaglyph is basically a post processing effect with OGL, and I don't have the skills to avoid _losing the possibly preset_ post processing effect after the user toggles the anaglyph mode off. I don't see this as much of a problem, since most stereoscopy users use a "proper" mode, like side-by-side, right?
In other words: the odds are low for a situation where a user wants to use anaglyph stereoscopy _and_ no stereoscopy with a post processing effect in a same gaming session.
And you can still use pps's with other stereoscopy modes, if you like.

I don't have NVIDIA hardware, so 3D Vision Hotkey has not been tested, but I suppose it should work.

### CODE REVIEW

File: Source/Core/DolphinWX/Frame.cpp [hunk 1]
```diff
@@ -1426,6 +1426,64 @@ void CFrame::ParseHotkeys()
 	{
 		State::Load(g_saveSlot);
 	}
+	if (IsHotkey(HK_TOGGLE_STEREO_SBS))
+	{
+		if (g_Config.iStereoMode != STEREO_SBS)
+		{
+			if (g_Config.sPostProcessingShader == "dubois")
```
Comment (degasus) [inline Source/Core/DolphinWX/Frame.cpp]: SBS and up/down don't require a postprocessing shader, so why do you toggle it?

File: Source/Core/DolphinWX/Frame.cpp [hunk 2]
```diff
@@ -1426,6 +1426,64 @@ void CFrame::ParseHotkeys()
...
+		{
+			g_Config.iStereoMode = STEREO_OFF;
+		}
+	}
+	if (IsHotkey(HK_TOGGLE_STEREO_ANAGLYPH))
+	{
+		if (g_Config.iStereoMode != STEREO_ANAGLYPH)
+		{
+			g_Config.iStereoMode = STEREO_ANAGLYPH;
+			g_Config.sPostProcessingShader = "dubois";
```
Comment (degasus) [inline Source/Core/DolphinWX/Frame.cpp]: Is there a way to just call PostProcessingShaderConfiguration::LoadShader instead of hardcoding a shader name?

File: Source/Core/DolphinWX/Frame.cpp [hunk 3]
```diff
@@ -1426,6 +1426,64 @@ void CFrame::ParseHotkeys()
...
+		{
+			g_Config.iStereoMode = STEREO_OFF;
+			g_Config.sPostProcessingShader = "";
+		}
+	}
+	if (IsHotkey(HK_TOGGLE_STEREO_3DVISION))
+	{
+		if (g_Config.iStereoMode != STEREO_3DVISION)
+		{
+			if (g_Config.sPostProcessingShader == "dubois")
```
Comment (degasus) [inline Source/Core/DolphinWX/Frame.cpp]: 3d vision also don't need a shader afaik

File: Source/Core/DolphinWX/Frame.cpp (hunk 1 above)
Comment (TurboK234) [inline Source/Core/DolphinWX/Frame.cpp]: The reason I included the postprocessing shader (specifically anaglyph "dubois") in these other hotkeys was the fact that just enabling a stereoscopy mode via g_config.iStereoMode does not disable the anaglyph pp shader. This lead to side-by-side and top/bottom stereoscopy with false anaglyph colors.
It's an assurement that only one stereoscopy mode is selected when you click the hotkey. And it only disables the anaglyph mode, other pps's are available and unaffected.

File: Source/Core/DolphinWX/Frame.cpp (hunk 2 above)
Comment (TurboK234) [inline Source/Core/DolphinWX/Frame.cpp]: This is the kind of question I was afraid of. I'm really a novice at C++, the fact I got this working in the first place was tha fact that VS 2013 has exellent search capabilities and I figured out that setting the anaglyph shader in Graphics/Settings in fact sets "dubois" in a similar manner as here.
This also seemed a rather straightforward way? But if you can think of another or a more abstract way to set the stereoscopy mode & required shader please tell ;)

File: Source/Core/DolphinWX/Frame.cpp (hunk 3 above)
Comment (TurboK234) [inline Source/Core/DolphinWX/Frame.cpp]: Same as with SBS and top/bottom, only assuring here that anaglyph is not left on when selecting 3D Vision.

**Comment by TurboK234** [comment]
So, I added two commits (one to fix a dissappearing ampersand "&" from the a hotkey description and another to comment the logic in the hotkeys), followed a guide to squash the commits and rebased everything with the master.
This PR could be further developed in (at least) two ways:
1) Storing the (possible previously set) post-processing shader whenever anaglyph is turned on (store as in declaring a new variable to do it), so it could be retrieved when anaglyph is turned off. This is not the way GUI works ATM, either.
2) Using an abstractified form of the shaders to set the correct one for anaglyph, instead of using the literal name.

Both are well beyond my know-how. I'm good with this myself. If this requires something more then it's fine if it gets closed. Someone can adopt this as a basis for a better implementation (or write something waaay better).

**Comment by aserna3** [comment]
@dolphin-emu-bot rebuild

**Comment by Tilka** [comment]
Please rebase on master instead of merging it.

**Comment by TurboK234** [comment]
Yeah I just noticed that resulting commit is a mess, hang on.

**Comment by TurboK234** [comment]
There, rebased, looks much cleaner.
